src/rlang/vec.h

```diff
@@ -77,6 +77,58 @@ sexp* const * r_list_deref_const(sexp* x) {
 #endif
 }
 
+static inline
+void* r_vec_deref0(enum r_type type, sexp* x) {
+  switch (type) {
+  case r_type_logical: return r_lgl_deref(x);
+  case r_type_integer: return r_int_deref(x);
+  case r_type_double: return r_dbl_deref(x);
+  case r_type_complex: return r_cpl_deref(x);
+  case r_type_raw: return r_raw_deref(x);
+  default: r_stop_unimplemented_type("r_vec_deref", type);
+  }
+}
+static inline
+void* r_vec_deref(sexp* x) {
+  return r_vec_deref0(r_typeof(x), x);
+}
+
+static inline
+const void* r_vec_deref_const0(enum r_type type, sexp* x) {
+  switch (type) {
+  case r_type_logical: return r_lgl_deref_const(x);
+  case r_type_integer: return r_int_deref_const(x);
+  case r_type_double: return r_dbl_deref_const(x);
+  case r_type_complex: return r_cpl_deref_const(x);
+  case r_type_raw: return r_raw_deref_const(x);
+  case r_type_character: return r_chr_deref_const(x);
+  case r_type_list: return r_list_deref_const(x);
+  default: r_stop_unimplemented_type("r_vec_deref_const", type);
+  }
+}
+static inline
+const void* r_vec_deref_const(sexp* x) {
+  return r_vec_deref_const0(r_typeof(x), x);
+}
+
+static inline
+int r_vec_elt_sizeof0(enum r_type type) {
```

Comment:
`r_type_sizeof()`?

Reply:
oh yes

Reply:
Though that only makes sense for vectors and maybe it's best to keep the name symmetry.